scylla-cdc-driver3/src/main/java/com/scylladb/cdc/cql/driver3/Driver3SchemaBuilder.java

```diff
@@ -12,125 +13,477 @@
+         * inconsistencies:
+         * <table border="1">
+         *     <caption>Proof of correctness</caption>
+         *     <tr>
+         *         <th>Inconsistency type</th>
+         *         <th>Schema change type</th>
+         *         <th>Commentary</th>
+         *     </tr>
+         *     <tr>
+         *         <td>Base schema newer than CDC log schema</td>
+         *         <td>Column added</td>
+         *         <td>This method will never be called for that added column,
+         *             as the column name was not present in the CDC log schema (caller).</td>
+         *     </tr>
+         *     <tr>
+         *         <td>Base schema newer than CDC log schema</td>
+         *         <td>Column removed</td>
+         *         <td>You cannot remove columns from the primary key. Therefore,
+         *             the removed column must have been a regular column. This
+         *             method will correctly return REGULAR for that column
+         *             (the type of that column as of the CDC schema time),
+         *             as it won't find this column in a primary key of the
+         *             base table.</td>
+         *     </tr>
+         *     <tr>
+         *         <td>Base schema newer than CDC log schema</td>
+         *         <td>Column renamed</td>
+         *         <td>Only clustering key columns can be renamed. We detect
+         *             if all clustering key columns in the base table
+         *             schema are present in the CDC log schema
+         *             (base schema: pk1, ck_new, col1;
+         *             CDC log schema: pk1, ck_old, col1). If there is
+         *             an inconsistency, that means that a clustering column
+         *             was renamed. That means that all other columns are not
+         *             changed. If the queried CDC column is a part of partition key
+         *             of base schema, that means that it was a partition
+         *             key (as of CDC schema time) - partition columns
+         *             were not modified (clustering column was modified).
+         *             Similarly for regular columns. But, if the column
+         *             is missing from the base table schema, that means
+         *             that we are seeing a old clustering column.
+         *             </td>
+         *     </tr>
+         *     <tr>
+         *         <td>Base schema newer than CDC log schema</td>
+         *         <td>Column data type changed</td>
+         *         <td>The column data type change does not change
+         *             the column set (column names, whether the columns
+         *             are in the primary key). Therefore, the schemas
+         *             are consistent with regard to column type.</td>
+         *     </tr>
+         *     <tr>
+         *         <td>CDC log schema newer than base schema</td>
```

Comment:
How is this possible for CDC log schema to be newer than base table schema?

Reply:
Duplicate of previous comment.